addons/markers/functions/fnc_onButtonClickConfirm.sqf

```diff
@@ -17,12 +17,27 @@
 params ["_buttonOk"];
 
 private _display = ctrlParent _buttonOk;
-private _description = _display displayctrl IDC_INSERT_MARKER;
+private _description = _display displayCtrl IDC_INSERT_MARKER;
 private _aceTimestamp = _display displayCtrl IDC_ACE_INSERT_MARKER_TIMESTAMP;
 
 // handle timestamp
 if (cbChecked _aceTimestamp && {ACE_player call FUNC(canTimestamp)}) then {
-    private _time = daytime;
+    private _time = switch (GVAR(timestampTimezone)) do {
+        case 1: {
+            systemTime params ["", "", "", "_hour", "_min", "_sec"];
+            _hour + _min/60 + _sec/3600
+        };
+        case 2: {
+            systemTimeUTC params ["", "", "", "_hour", "_min", "_sec"];
+            _hour = _hour + round (GVAR(timestampUTCOffset));
+            _hour = _hour < 0 ? _hour + _hour % 24 : _hour % 24;
+            _hour = if (Hour < 0) then { 24 + Hour };
```

Comment:
```suggestion
            _hour = _hour + round ((GVAR(timestampUTCOffset) % 24) + 24 % 24);
            _hour = _hour % 24;
```

Reply:
Modulo goes before adding and subtraction right?

Reply:
Implemented this with a simular approach

Reply:
I tested it rather thoroughly, and it worked.

Reply:
```sqf
_hour = _hour + round ((GVAR(timestampUTCOffset) % 24) + 24 % 24);
_hour = _hour % 24;
```

can be simplified to:
```sqf
_hour = (_hour + round (GVAR(timestampUTCOffset))) % 24;
```

so it's just the same as only using `_hour = _hour % 24`.

It has the same problem with mod returning just the negative value of the positive mod.

![20231113215850_1](https://github.com/acemod/ACE3/assets/78355193/7308c7a0-e316-4e57-b576-945d86e7be88)

This would be the best approach in my opinion:

```sqf
_hour = (_hour + round (GVAR(timestampUTCOffset))) % 24;
_hour = if (_hour < 0) then { 24 + _hour } else { _hour };
```

Reply:
> ```sqf
> _hour = _hour + round ((GVAR(timestampUTCOffset) % 24) + 24 % 24);
> _hour = _hour % 24;
> ```
> 
> can be simplified to:
> 
> ```sqf
> _hour = (_hour + round (GVAR(timestampUTCOffset))) % 24;
> ```

No, it can't, because as you pointed out yourself, it would be as `_hour = _hour % 24`.

You could simplify it to:
```sqf
_hour = ((_hour + round GVAR(timestampUTCOffset)) % 24) + 24 % 24;
```

You remove all multiples of 24 with the inner most `%`. You then add 24 to make sure it's positive, but you ensure it's not larger than 24 by taking the modulo.

Seeing as you implemented it in the latest commit means it's been resolved.